### ISSUE TITLE
There are differences in phynionet reproduction accuracy

### ISSUE THREAD

**Issue by tangxiaoyue107** [usail-hkust/t-PatchGNN#9]
Since the active and ushcn authors uploaded the pt file, the running results i trained are accurate with the paper. However, the pt file of the physinet dataset is not given by the author, and is preprocessed by Physinet.py. My preprocessing results and logs are placed in the following package. However, the accuracy of pyhnionet's runnig results is not as good as described in this paper. Can authors upload pt data sets pre-processed by physinet or share the training logs?
[logs.zip](https://github.com/user-attachments/files/17428598/logs.zip)

**Comment by albert-y1n** [comment]
Due to the large size of the processed PhysioNet data, we cannot upload it directly to GitHub.

For any discrepancies in reproduction results, ensure the processed data is labeled as 'set-x_0.0.pt' and verify the 'quantization' parameter. Additionally, please refer to our provided scripts for hyperparameter details.

**Comment by willzhang3** [comment]
> Since the active and ushcn authors uploaded the pt file, the running results i trained are accurate with the paper. However, the pt file of the physinet dataset is not given by the author, and is preprocessed by Physinet.py. My preprocessing results and logs are placed in the following package. However, the accuracy of pyhnionet's runnig results is not as good as described in this paper. Can authors upload pt data sets pre-processed by physinet or share the training logs? [logs.zip](https://github.com/user-attachments/files/17428598/logs.zip)

Based on your logs, it seems that your hyper-parameters, such as lr, tf_layer, patience, and quantization, are different from ours. Please entirely follow our provided code and scripts to ensure an identical setting for the reproduction of results.

**Comment by tangxiaoyue107** [comment]
> > Since the active and ushcn authors uploaded the pt file, the running results i trained are accurate with the paper. However, the pt file of the physinet dataset is not given by the author, and is preprocessed by Physinet.py. My preprocessing results and logs are placed in the following package. However, the accuracy of pyhnionet's runnig results is not as good as described in this paper. Can authors upload pt data sets pre-processed by physinet or share the training logs? [logs.zip](https://github.com/user-attachments/files/17428598/logs.zip)
> 
> Based on your logs, it seems that your hyper-parameters, such as lr, tf_layer, patience, and quantization, are different from ours. Please entirely follow our provided code and scripts to ensure an identical setting for the reproduction of results.


> Due to the large size of the processed PhysioNet data, we cannot upload it directly to GitHub.
> 
> For any discrepancies in reproduction results, ensure the processed data is labeled as 'set-x_0.0.pt' and verify the 'quantization' parameter. Additionally, please refer to our provided scripts for hyperparameter details.


I am sorry that I have not noticed the quantization parameters before, thank you for pointing it out! May I ask what quantity of quantization you have set? 0.0? py after I tried your physionet.py to generate st-x_0.0.py file, I used the parameter Settings in your script again, but there was still a certain gap with the reproduced data in the original paper. I have uploaded the log file, could you please check the reason again?  Thank you very much for your work and reply to me, which is of great help to my study!
[logs.zip](https://github.com/user-attachments/files/17454695/logs.zip)

**Comment by willzhang3** [comment]
[physionet_tPatchGNN_def_8.0patch_8.0stride_1layer_0.001lr.log.zip](https://github.com/user-attachments/files/17471892/physionet_tPatchGNN_def_8.0patch_8.0stride_1layer_0.001lr.log.zip)

We just re-run this repository and obtained identical results to our paper. We recommend trying to keep your experimental configurations the same as instructions in README, including cuda version (12.2), libraries versions, etc.

**Comment by willzhang3** [comment]
> > > Since the active and ushcn authors uploaded the pt file, the running results i trained are accurate with the paper. However, the pt file of the physinet dataset is not given by the author, and is preprocessed by Physinet.py. My preprocessing results and logs are placed in the following package. However, the accuracy of pyhnionet's runnig results is not as good as described in this paper. Can authors upload pt data sets pre-processed by physinet or share the training logs? [logs.zip](https://github.com/user-attachments/files/17428598/logs.zip)
> > 
> > 
> > Based on your logs, it seems that your hyper-parameters, such as lr, tf_layer, patience, and quantization, are different from ours. Please entirely follow our provided code and scripts to ensure an identical setting for the reproduction of results.
> 
> > Due to the large size of the processed PhysioNet data, we cannot upload it directly to GitHub.
> > For any discrepancies in reproduction results, ensure the processed data is labeled as 'set-x_0.0.pt' and verify the 'quantization' parameter. Additionally, please refer to our provided scripts for hyperparameter details.
> 
> I am sorry that I have not noticed the quantization parameters before, thank you for pointing it out! May I ask what quantity of quantization you have set? 0.0? py after I tried your physionet.py to generate st-x_0.0.py file, I used the parameter Settings in your script again, but there was still a certain gap with the reproduced data in the original paper. I have uploaded the log file, could you please check the reason again? Thank you very much for your work and reply to me, which is of great help to my study! [logs.zip](https://github.com/user-attachments/files/17454695/logs.zip)

We have provided the processed dataset of PhysioNet. You may use it to ease the reproduction.

**Comment by tangxiaoyue107** [comment]
After using your original pt file, I was able to get the results of physionet. Thank you very much! To some extent, this shows that I am not skilled enough in preprocessing code. Thank you very much for your patience and reply! Your work has helped me a lot in my study! Hope to see more of your work on irregular timing series forecasting in the future!

**Comment by willzhang3** [comment]
We are glad that you have successfully reproduced the results :)

**Comment by Csq-0310** [comment]
> 由于现役作者和USHCN作者上传了pt文件，我训练的运行结果与论文一致。然而，physinet数据集的pt文件并非作者提供，而是由 Physinet.py 预处理的。我的预处理结果和日志被放在以下资料包中。然而，pyhnionet的runnig结果的准确性并不像本文描述的那么好。作者可以上传physinet预处理的病人数据集或分享训练日志吗？ [logs.zip](https://github.com/user-attachments/files/17428598/logs.zip)

Hello, I'm encountering the same issue. Could you please share how you resolved it in the end

**Comment by Csq-0310** [comment]
Dear Authors,
I am currently reproducing your code and have referred to your result logs. I am using the exact same parameters as specified, but I am encountering the same issue. Could you please provide specific solutions to resolve this? I would be extremely grateful for your response.
Attached is my running log for your reference.

[logs.txt](https://github.com/user-attachments/files/23761878/logs.txt)

**Comment by willzhang3** [comment]
@Csq-0310  Please directly use the processed .pt file to reproduce the results. We have provided this file for PhysioNet.

**Comment by Csq-0310** [comment]
Thank you for your reply. I have re-downloaded and run the latest code, and also used the newly processed .pt files. However, the results still don't seem very satisfactory.

[log.txt](https://github.com/user-attachments/files/23784912/log.txt)

**Comment by willzhang3** [comment]
@Csq-0310  Do you try more random seeds?  It seems that your result has not converged very well.

**Comment by Csq-0310** [comment]
Thank you for your reply.I have used five random seeds, and the average MSE (Mean Squared Error) is 0.00593, which shows a slight gap compared with the 0.00498 reported in the original paper. However, there is no significant difference in the MAE (Mean Absolute Error) metric.

**Comment by willzhang3** [comment]
@Csq-0310  Hi, are you sure that you have decompressed [processed.zip](https://github.com/usail-hkust/t-PatchGNN/blob/main/data/physionet/processed.zip) and use the .pt file in your experiments? Instead, the code will process the raw data based on your experimental environments, which may lead to a slight discrepancy in the processed data.

**Comment by Csq-0310** [comment]
Hello, I have been using the extracted set-x.00.pt file. I have also increased the value of patience to 20, but the performance still seems unsatisfactory. This paper is very important to me—would it be possible to exchange contact information for further communication? Below is my dataset file.

**Comment by willzhang3** [comment]
@Csq-0310  It seems your dataset file was not successfully uploaded.  Can you provide your environment configuration and the full log.txt file? 

Moreover, you can contact me via email (vegazhang3@gmail.com) if you think this would enable a more effective discussion.
